Src/GoogleApis.Auth.DotNet4.Tests/OAuth2/DefaultCredentialProvider/DefaultCredentialProviderTests.cs

```diff
@@ -0,0 +1,271 @@
+        { 
+            try
+            {
+                var credential = await credentialProvider.GetDefaultCredentialAsync();
+                Assert.Fail();
+            }
+            catch (InvalidOperationException e)
+            {
+                Assert.IsTrue(e.Message.Contains("The Application Default Credentials are not available"));
+            }
+        }
+
+        /// <summary>Environment variable points to a non existant credential file.</summary>
+        [Test]
+        public async Task GetDefaultCredential_MissingCredentialFile()
+        {
+            credentialProvider.SetEnvironmentVariable(CredentialEnvironmentVariable, WellKnownCredentialFilePath);
+            
+            try
+            {
+                var credential = await credentialProvider.GetDefaultCredentialAsync();
+                Assert.Fail();
+            }
+            catch (InvalidOperationException e)
+            {
+                Assert.IsTrue(e.Message.Contains("Please check the value of the Environment Variable GOOGLE_APPLICATION_CREDENTIALS"));
+            }
+        }
+
+        /// <summary>Credential file has invalid content format.</summary>
+        [Test]
+        public async Task GetDefaultCredential_InvalidCredentialFile()
+        {
+            string credentialFilepath = "TempFilePath.json";
+            credentialProvider.SetEnvironmentVariable(CredentialEnvironmentVariable, credentialFilepath);
+            credentialProvider.SetFileContents(credentialFilepath, "Invalid Credentials File Contents");
+
+            try
+            {
+                var credential = await credentialProvider.GetDefaultCredentialAsync();
+                Assert.Fail();
+            }
+            catch (InvalidOperationException e)
+            {
+                Assert.IsTrue(e.Message.Contains("Error reading credential file from location"));
+            }
+        }
+
+        /// <summary>Credential file contains invalid PKCS8 key.</summary>
+        [Test]
+        public async Task GetDefaultCredential_BrokenPkcs8Key()
+        {
+            string credentialFilepath = "TempFilePath.json";
+            credentialProvider.SetEnvironmentVariable(CredentialEnvironmentVariable, credentialFilepath);
+            credentialProvider.SetFileContents(credentialFilepath, BrokenPkcs8KeyServiceAccountCredentialFileContents);
+
+            try
+            {
+                var credential = await credentialProvider.GetDefaultCredentialAsync();
+                Assert.Fail();
+            }
+            catch (InvalidOperationException e)
+            {
+                Assert.IsTrue(e.Message.Contains("Error reading credential file from location"));
+            }
+        }
+
+        // TODO(jtattermusch): test compute engine credential.
+
+        #endregion
+    }
+
+    /// <summary>A mock for the <see cref="Google.Apis.Auth.OAuth2.DefaultCredentialProvider"/>.</summary>
+    class TestDefaultCredentialProvider : DefaultCredentialProvider
```

Comment:
I would put it at the beginning of the file, just because I did in other tests files as well...